extension/otto/gohan_core.go

```diff
@@ -141,23 +141,32 @@ func init() {
 		}
 
 		err = env.Load("<Gohan built-ins>", `
-		var gohan_handler = {}
+		var gohan_handler = {};
+		var gohan_caller = "";
 		function gohan_register_handler(event_type, func){
 		  if(_.isUndefined(gohan_handler[event_type])){
 		    gohan_handler[event_type] = [];
 		  }
-		  gohan_handler[event_type].push(func)
+		  var id = gohan_uuid();
+		  gohan_handler[event_type].push({fn:func,id:id})
+		  gohan_log_debug("gohan built-in: register handler: type = '" + event_type.toString() +
+				"'; index = " + (gohan_handler[event_type].length - 1).toString() +
+				"'; id = " + id +
+				"; handler = '" + func.toString().replace(/\s\s+/g, '') + "'")
 		}
-
 		function gohan_handle_event(event_type, context){
 		  if(_.isUndefined(gohan_handler[event_type])){
 		    return;
 		  }
-
 		  for (var i = 0; i < gohan_handler[event_type].length; ++i) {
 		    try {
 		      var old_module = gohan_log_module_push(event_type);
-		      gohan_handler[event_type][i](context);
+		      gohan_caller = gohan_handler[event_type][i].id;
```

Comment:
It's the same problem as with marking DB logs with handler's id. I suggest that this may be a future improvement.